### PR TITLE
[Merged by Bors] - feat: optionEquiv lemmas

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds lemmas about what `optionEquivRight`/`optionEquivLeft` evaluates to.

---

- [x] depends on: #11193

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/MvPolynomial/Equiv.lean [hunk 1]
```diff
@@ -306,6 +306,22 @@ def optionEquivRight : MvPolynomial (Option S₁) R ≃ₐ[R] MvPolynomial S₁
...
+@[simp]
+lemma optionEquivRight_X_none : optionEquivRight R S₁ (X none) = C Polynomial.X := by
+  unfold optionEquivRight AlgEquiv.ofAlgHom
+  simp only [AlgEquiv.coe_mk, aeval_X, Option.elim]
+
+@[simp]
+lemma optionEquivRight_C (r : R) : optionEquivRight R S₁ (C r) = C (Polynomial.C r) := by
+  unfold optionEquivRight AlgEquiv.ofAlgHom
+  simp only [Option.elim, AlgEquiv.coe_mk, aeval_C]
+  rfl
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/MvPolynomial/Equiv.lean:323]:
```suggestion
  simp only [Option.elim, AlgEquiv.coe_mk, aeval_C, C_eq_algebraMap, algebraMap_apply]
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Bonus points if you do the same for optionEquivLeft

**Comment by BoltonBailey** [comment]
Ok done, but I don't know how to remove the rfl from the last one like you did.

**Comment by BoltonBailey** [comment]
Ok I did this but I guess simp already proves some of those lemmas.

Looking into this I am a little mystified. We have this relevant lemma [Polynomial.C_eq_algebraMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/AlgebraMap.html#Polynomial.C_eq_algebraMap). But shouldn't that lemma be turned around and marked as `simp`?

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**